terraform/terraform_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'spec_helper'
+require 'docker'
+require 'serverspec'
+
+describe "Terraform image" do
+    before(:all) {
+        set :docker_image, find_image_id('terraform:latest')
+    }
+
+    it "installs the Alpine linux" do
+        expect(file("/etc/alpine-release")).to be_file
+    end
+
+    it "installs Root Certificates" do
+        expect(file("/usr/share/ca-certificates/mozilla/GlobalSign_Root_CA.crt")).to be_file
+    end
+
+    it "checks if terraform binary is executable" do
+        expect(file("/usr/local/bin/terraform")).to be_mode 755
+    end
+
+    it "installs all the whitelisted binaries" do
+        command("echo $BINARY_WHITELIST").stdout.split.each { |f|
+            expect(file("/usr/local/bin/#{f}")).to be_mode 755
+        }
```

Comment:
This test feels a little circular to me. As it stands, nothing will fail if one of the required binaries is removed in the `Dockerfile`.

This is effectively testing that the `Dockerfile` installs what it says it's going to, which is a different thing from testing that it installs the things we need.